src/operations/support/editorUtils.ts

```diff
@@ -1,55 +1,71 @@
 import { HandlerContext } from "../../HandlerContext";
-import { ActionResult } from "../../internal/util/ActionResult";
-import { GitCommandGitProject } from "../../project/git/GitCommandGitProject";
 import { GitProject } from "../../project/git/GitProject";
-import { RepoId } from "../common/RepoId";
-import { ProjectEditor } from "../edit/projectEditor";
+import { Project } from "../../project/Project";
+import { defaultRepoLoader } from "../common/defaultRepoLoader";
+import { isRepoId, RepoId } from "../common/RepoId";
+import { RepoLoader } from "../common/repoLoader";
+import {
+    CommitInfo, EditInfo, isCommitInfo, isCustomExecutionEditInfo, isPullRequestInfo,
+    PullRequestInfo,
+} from "../edit/editModes";
+import { EditResult, ProjectEditor, successfulEdit } from "../edit/projectEditor";
 
 /**
- * Edit a GitHub project using a PR
+ * Edit a GitHub project using a PR or branch
  * @param token GitHub token
  * @param context handler context for this operation
  * @param repo repo id
  * @param editor editor to use
- * @param pr structure of the PR
+ * @param repoLoader repo loading strategy
+ * @param ei how to persist the edit
  */
-export function editUsingPullRequest(token: string,
-                                     context: HandlerContext,
-                                     repo: RepoId,
-                                     editor: ProjectEditor<any>,
-                                     pr: PullRequestInfo): Promise<ActionResult<GitProject>> {
-    console.log("Editing project " + JSON.stringify(repo));
-    return GitCommandGitProject.cloned(token, repo.owner, repo.repo)
-        .then(gp => editProjectUsingPullRequest(context, repo, gp, editor, pr));
+export function loadAndEditRepo(token: string,
+                                context: HandlerContext,
+                                repo: RepoId | Project,
+                                editor: ProjectEditor,
+                                ei: EditInfo,
+                                repoLoader: RepoLoader =
+                                    defaultRepoLoader(token)): Promise<EditResult> {
+    const loadRepo: Promise<Project> = isRepoId(repo) ? repoLoader(repo) : Promise.resolve(repo);
+    if (isPullRequestInfo(ei)) {
+        return loadRepo.then(gp => editProjectUsingPullRequest(context, gp as GitProject, editor, ei));
+    } else if (isCommitInfo(ei)) {
+        return loadRepo.then(gp => editProjectUsingBranch(context, gp as GitProject, editor, ei));
+    } else if (isCustomExecutionEditInfo(ei)) {
+        return loadRepo.then(ei.edit);
+    } else {
+        // No edit to do
```

Comment:
is this masking errors? Seems like there'd be a "nevermind" EditMode and anything completely unexpected would be an error.